test/abstractarray.jl

```diff
@@ -389,48 +389,52 @@ Base.iteratorsize{N}(::Type{GenericIterator{N}}) = Base.SizeUnknown()
 
 function test_map(::Type{TestAbstractArray})
 
-    for typ in (Float16, Float32, Float64,
-                Int8, Int16, Int32, Int64, Int128,
-                UInt8, UInt16, UInt32, UInt64, UInt128
-    ),
-        arg_typ in (Integer,
-                    Signed,
-                    Unsigned
-    )
-        X = typ[1:10...]
-        _typ = typeof(arg_typ(one(typ)))
-        @test map(arg_typ, X) == _typ[1:10...]
-    end
+    empty_pool = WorkerPool()
+    pmap_fallback = (f, c...) -> pmap(empty_pool, f, c...)
+
+    for mapf in [map, asyncmap, pmap_fallback]
+
+        for typ in (Float16, Float32, Float64,
+                    Int8, Int16, Int32, Int64, Int128,
+                    UInt8, UInt16, UInt32, UInt64, UInt128
+        ),
+            arg_typ in (Integer,
+                        Signed,
+                        Unsigned
+        )
+            X = typ[1:10...]
+            _typ = typeof(arg_typ(one(typ)))
+            @test mapf(arg_typ, X) == _typ[1:10...]
+        end
 
-    # generic map
-    f(x) = x + 1
-    I = GenericIterator{10}()
-    @test map(f, I) == Any[2:11...]
-    @test collect(Base.StreamMapIterator(f, I)) == Any[2:11...]
-
-    # AbstractArray map for 2 arg case
-    f(x, y) = x + y
-    B = Float64[1:10...]
-    C = Float64[1:10...]
-    @test map(f, convert(Vector{Int},B), C) == Float64[ 2 * i for i in 1:10 ]
-    @test map(f, Int[], Float64[]) == Union{}[]
-    @test collect(Base.StreamMapIterator(f, Int[], Float64[])) == Float64[]
-    # map with different result tyoes
-    let m = map(x->x+1, Number[1, 2.0])
-        @test isa(m, Vector{Real})
-        @test m == Real[2, 3.0]
-    end
+        # generic map
+        f(x) = x + 1
+        I = GenericIterator{10}()
+        @test mapf(f, I) == Any[2:11...]
+
+        # AbstractArray map for 2 arg case
+        f(x, y) = x + y
+        B = Float64[1:10...]
+        C = Float64[1:10...]
+        @test mapf(f, convert(Vector{Int},B), C) == Float64[ 2 * i for i in 1:10 ]
+        @test mapf(f, Int[], Float64[]) == Union{}[]
+        # map with different result tyoes
```

Comment:
The typo was there already, I didn't notice (I'm a bit dyslexic), but I'll push a fix for it.